librz/asm/arch/evm/evmdis.c

```diff
@@ -0,0 +1,374 @@
+		case 0x5a:
+			rz_str_cpy(buf_asm, "GAS") break;
+		case 0x5b:
+			rz_str_cpy(buf_asm, "JUMPDEST") break;
+		case 0x60:
+			instruction_size = 2;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH1 0x%s", opcode); break;
+		case 0x61:
+			instruction_size = 3;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH2 0x%s", opcode); break;
+		case 0x62:
+			instruction_size = 4;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH3 0x%s", opcode); break;
+		case 0x63:
+			instruction_size = 5;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH4 0x%s", opcode); break;
+		case 0x64:
+			instruction_size = 6;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH5 0x%s", opcode); break;
+		case 0x65:
+			instruction_size = 7;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH6 0x%s", opcode); break;
+		case 0x66:
+			instruction_size = 8;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH7 0x%s", opcode); break;
+		case 0x67:
+			instruction_size = 9;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH8 0x%s", opcode); break;
+		case 0x68:
+			instruction_size = 10;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH9 0x%s", opcode); break;
+		case 0x69:
+			instruction_size = 11;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH10 0x%s", opcode); break;
+		case 0x6a:
+			instruction_size = 12;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH11 0x%s", opcode); break;
+		case 0x6b:
+			instruction_size = 13;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH12 0x%s", opcode); break;
+		case 0x6c:
+			instruction_size = 14;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH13 0x%s", opcode); break;
+		case 0x6d:
+			instruction_size = 15;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH14 0x%s", opcode); break;
+		case 0x6e:
+			instruction_size = 16;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH15 0x%s", opcode); break;
+		case 0x6f:
+			instruction_size = 17;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH16 0x%s", opcode); break;
+		case 0x70:
+			instruction_size = 18;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH17 0x%s", opcode); break;
+		case 0x71:
+			instruction_size = 19;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH18 0x%s", opcode); break;
+		case 0x72:
+			instruction_size = 20;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
+			rz_strf(buf_asm, "PUSH19 0x%s", opcode); break;
+		case 0x73:
+			instruction_size = 21;
+			rz_hex_bin2str(buf+1, instruction_size-1, opcode);
```

Comment:
You need to always check if the buffer is big enough. you cannot just read it as is and assume it will always be.